.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,55 @@
+name: "CodeQL"
+
+on:
+  workflow_call:
+
+jobs:
+  codeql-analyze:
+    name: CodeQL Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go' ]
```

Comment:
Are we planning to use this for other languages? Are we going to add the languages to the matrix? Should we consider using an input variable here with `go` as default?

Reply:
Yes, we are planning on using this for other languages. However, when I tried to do `[ ${{ inputs.language }} ]`, the workflow fails due to syntax error. So, for now I'll leave it as `'go'`, and I'll try and figure out how to get that working later on once I'm actually adding another language.

I'd like to figure out how to add multiple languages (in case a repo has multiple languages). But this also failed in my testing. Would need to spend more time on it - a bunch more sleuthing on the internet.

Reply:
btw, when I used an input for testing I did set `'go'` as the default. So I will use that method once we figure out how to get the syntax working.

Reply:
If we cannot templatize the language, we can always use a file with a different name,  `codeql-analysis-javascript.yml`